corehq/apps/app_manager/models.py

```diff
@@ -4966,14 +4966,19 @@ def wrap(cls, data):
         if 'original_doc' in data:
             data['copy_history'] = [data.pop('original_doc')]
             should_save = True
+        return should_save
 
+    @classmethod
+    def wrap(cls, data, scrap_old_conventions=True):
+        if scrap_old_conventions:
+            should_save = cls._scrap_old_conventions(data)
         data["description"] = data.get('description') or data.get('short_description')
 
         self = super(ApplicationBase, cls).wrap(data)
         if not self.build_spec or self.build_spec.is_null():
             self.build_spec = get_default_build_spec()
 
-        if should_save:
+        if scrap_old_conventions and should_save:
```

Comment:
Looks like that was some sort of migration. I have seen that around and i guess its something that is over now.
Can we just run it over all application docs once and clean the code up? Just in case if you are able to understand the context of it better than i did because you came 6 months before me to CommCare :)

Reply:
This is a classic "lazy migration" pattern that we thankfully seem to have dropped (note: [I'm not innocent in all this](https://github.com/dimagi/commcare-hq/commit/9e70aa20572edcbfb3ca15047fb38ad50c1e4efd#diff-535390eb5c83a4cd3449f82afd1fa9fbR3682)).  The way it worked was:

1. Make a change that requires a migration
1. Modify the `.wrap()` method to handle old data and convert it to new data
1. As objects are loaded, they are converted to the new format
1. You're never sure whether all objects have been loaded at some point
1. 3 years later someone gets frustrated enough with the code to actually write the migration that should've happened initially, or at least they load and save all documents of that type.
1. The scaffolding is removed

This problem is what triggered me to write [`iter_update`](https://github.com/dimagi/commcare-hq/blob/3e57086c3870c31f1bbf9f32b70db0f374369c6e/corehq/util/couch.py#L280-L280).  That should make it pretty easy to run basic updates like these proactively and performantly if you're ever feeling magnanimous.

Reply:
haha, being honest but still coming out as the hero who wants to save the world :)